internal/cloudsql/instance.go

```diff
@@ -185,11 +184,10 @@ func (i *Instance) scheduleRefresh(d time.Duration) *refreshResult {
 	res := &refreshResult{}
 	res.ready = make(chan struct{})
 	res.timer = time.AfterFunc(d, func() {
-		ctx, cancel := context.WithTimeout(context.Background(), i.refreshTimeout)
-		res.md, res.tlsCfg, res.expiry, res.err = performRefresh(ctx, i.client, i.clientLimiter, i.connName, i.key)
-		cancel()
-
+		ctx := context.Background() // TODO: store this in Instance
```

Comment:
Re: the TODO. Or Better pass it through from the client call site so clients could cancel the refresh.

Reply:
Feel free to weigh in here: https://github.com/kurtisvg/cloud-sql-go-connector/issues/5#issuecomment-834820781

Reply:
(I'm leaning towards opt3 right now)